modules/Microsoft.Compute/virtualMachines/deploy.bicep

```diff
@@ -306,11 +316,17 @@ var linuxConfiguration = {
     publicKeys: publicKeysFormatted
   }
   provisionVMAgent: provisionVMAgent
+  patchSettings: (provisionVMAgent && (patchMode =~ 'AutomaticByPlatform' || patchMode =~ 'ImageDefault')) ? {
+    patchMode: patchMode
+  } : null
 }
 
 var windowsConfiguration = {
   provisionVMAgent: provisionVMAgent
-  enableAutomaticUpdates: enableAutomaticUpdates
+  enableAutomaticUpdates: patchMode !~ 'Manual' ? enableAutomaticUpdates : false
+  patchSettings: (provisionVMAgent && (patchMode =~ 'AutomaticByPlatform' || patchMode =~ 'AutomaticByOS' || patchMode =~ 'Manual')) ? {
```

Comment:
Updates to this file are also part of the broader design principle discussion mentioned in PR #2454 

"Should we enforce the right behavior besides the user decision or should we allow the deployment to fail if parameters are not set correctly?". We often go with this last option. 

Also in this case I'm curious about other maintainers' thoughts @AlexanderSehr @ahmadabdalla

Reply:
What I've mentioned in another PR #2455, is that I would personally prefer the values being defined in a parameter file, are the values that are injected into the template. This way if we have multiple config files, it's easier for users to know what's the difference.

If we override what users put in their parameter files based on an automated logic, it can get confusing plus will require us to take on additional responsibilities to maintain the correction logic in our modules. If MS releases a new API version that has parameters that would conflict with others, then we will just keep adding more to that logic. I've dealt with this before and it can get complex sometimes.

I would prefer the template deployment to fail rather than testing for each toxic parameter combinations and call out the statement in the parameter description about values conflicting with others.

But that's just my view, if we're going with corrections we need to call it out in the parameter descriptions at least and then take on handling this logic for CARML customers. We already do that for some modules I think so it's a matter of setting the principle.

Reply:
I've removed this condition, and updated the description for the ``enableAutomaticUpdates` parameter, stating if `patchMode` is set to `Manual`, this parameter must be set to `false`